### ISSUE TITLE
The narrator does not announce the name of the checkbox descriptively 

### ISSUE THREAD

**Issue by shrinidhizzz** [microsoft/fluentui#14891]
### Environment Information

- Windows Version - Win10 
- Chromium Edge Version: 85.0.552.1 (Official Build) dev (64-bit)
- Chrome Version: 83.0.4103.116 (Official Build) (64-bit)
- NVDA Version: 2020.1

### Please provide a reproduction of the issue in a codepen:

Providing an isolated reproduction of the issue in a codepen makes it much easier for us to help you. Here are some ways to get started:

  * Go to https://codepen.io/ShrinidhiMicrosoft/pen/yLOPoVM?editors=1010 for a starter codepen
  * Turn on the screen reader tool.
  * Listen what the reader reads when you enter to open the dialog open.

Alternatively, you can also use https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist/basic to get permanent repro links if the repro occurs with an example.
(A permanent link is preferable to "use the website" as the website can change.)

#### Actual behavior:

The narrator does not announce the name of the checkbox in the table descriptively.

#### Expected behavior:

The narrator should announce each checkbox of the table more descriptively.

### Documentation describing expected behavior

![image](https://user-images.githubusercontent.com/30197247/92198874-26fdec80-ee93-11ea-99c5-3b96286f72ab.png)

**Comment by paulgildea** [comment]
@smhigley FYI

@miroslavstastny - This might be an issue with the authored sample less so the `DetailsList` itself - not sure.

@dzearing - What was the expectation on what the `DetailsList` provides by default and what a11y content does the author need to provide?

**Comment by dzearing** [comment]
@cschlechty or @ThomasMichon would probably be better to answer that.

**Comment by paulgildea** [comment]
@joschect Do you think that any of the recent `DetailsList` changes could have impacted this?
